File: core/src/main/java/net/siegerpg/siege/core/fishing/catches/loot/Sugar.java
package net.siegerpg.siege.core.fishing.catches.loot;

import net.siegerpg.siege.core.fishing.catches.Fish;

public class Sugar extends Fish {

	public Sugar() {

		super(40, 0.8, 14, 10,
		      new net.siegerpg.siege.core.items.implemented.misc.food.Sugar().getUpdatedItem(false).asQuantity(16)
		     );
	}

}

File: core/src/main/kotlin/net/siegerpg/siege/core/items/implemented/misc/food/Sugar.kt
package net.siegerpg.siege.core.items.implemented.misc.food

import net.siegerpg.siege.core.items.enums.Rarity
import net.siegerpg.siege.core.items.types.misc.CustomFood
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType

class Sugar() : CustomFood(
	name = "Sugar",
	customModelData = 330002,
	description = listOf("Gas GAS GASSS!!!", "Speed 3 for 30 seconds"),
	levelRequirement = 0,
	material = Material.TROPICAL_FISH,
                          ) {

	override fun speciality(player: Player) {
		val realPotion = PotionEffect(PotionEffectType.SPEED, 600, 2)
		player.addPotionEffect(realPotion)
	}

	constructor(quality: Int) : this() {
		this.quality = 100
		this.rarity = Rarity.LEGENDARY
		this.serialize()
	}

	constructor(item: ItemStack) : this() {
		this.item = item
		this.deserialize()
	}

}